enclave-tls/src/core/main.c

```diff
@@ -3,36 +3,58 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <strings.h>
+#include <stdlib.h>
+#include <string.h>
 #include <enclave-tls/log.h>
 #include <enclave-tls/err.h>
 #include "internal/core.h"
 #include "internal/tls_wrapper.h"
 #include "internal/enclave_quote.h"
 #include "internal/crypto_wrapper.h"
+#ifdef SGX
+#include "etls_t.h"
+#endif
 
 /* The global configurations present by /opt/enclave-tls/config.toml */
 etls_core_context_t global_core_context;
 /* The global log level used by log.h */
 enclave_tls_log_level_t global_log_level = ENCLAVE_TLS_LOG_LEVEL_DEFAULT;
 
+void etls_exit(void)
+{
+#ifndef SGX
+	exit(EXIT_FAILURE);
+#else
+	ocall_exit();
+#endif
+}
+
+#ifdef SGX
+void libenclave_tls_init(void)
+#else
 void __attribute__((constructor)) libenclave_tls_init(void)
+#endif
 {
 	ETLS_DEBUG("called\n");
 
-	char *log_level_str = getenv("ENCLAVE_TLS_GLOBAL_LOG_LEVEL");
+	char *log_level_str = NULL;
+#ifdef SGX
+	ocall_getenv("ENCLAVE_TLS_GLOBAL_LOG_LEVEL", log_level_str);
+#else
+	log_level_str = getenv("ENCLAVE_TLS_GLOBAL_LOG_LEVEL");
```

Comment:
It is better to define a new function called etls_utils_getenv() in a separate source file, and use the conditional compilation in there. This makes the codes clean, instead of using `ifdef SGX` on the fly. I believe this is a common issue in your codes. I call this issue pattern as UTILS. Anything I marked as UTILS needs to apply the same change.

Reply:
ok，some function use common instead，some not like this，i will modify unified.